src/ApiService/ApiService/onefuzzlib/NodeOperations.cs

```diff
@@ -168,14 +175,104 @@ IOnefuzzContext context
     public async Async.Task ReimageLongLivedNodes(Guid scaleSetId) {
         var timeFilter = $"not (initialized_at ge datetime'{(DateTimeOffset.UtcNow - INodeOperations.NODE_REIMAGE_TIME).ToString("o")}')";
 
-        await foreach (var node in QueryAsync($"(scaleset_id eq {scaleSetId}) and {timeFilter}")) {
+        await foreach (var node in QueryAsync($"(scaleset_id eq '{scaleSetId}') and {timeFilter}")) {
             if (node.DebugKeepNode) {
                 _logTracer.Info($"removing debug_keep_node for expired node. scaleset_id:{node.ScalesetId} machine_id:{node.MachineId}");
             }
             await ToReimage(node with { DebugKeepNode = false });
         }
     }
 
+
+    public async Async.Task MarkOutdatedNodes() {
+        //#if outdated agents are allowed, do not attempt to update
+        bool allowOutdatedAgent;
+        var parsed = bool.TryParse(_context.ServiceConfiguration.OneFuzzAllowOutdatedAgent ?? "false", out allowOutdatedAgent);
+        if (parsed && allowOutdatedAgent) {
+            return;
+        }
+
+        var outdated = this.SearchOutdated(excludeUpdateScheduled: true);
+        await foreach (var node in outdated) {
+            _logTracer.Info($"node is outdated: {node.MachineId} - node_version:{node.Version} api_version:");
+
+            if (node.Version == "1.0.0") {
+                await ToReimage(node, done: true);
+            } else {
+                await ToReimage(node);
+            }
+        }
+    }
+
+
+    public static string SearchOutdatedQuery(
+    string oneFuzzVersion,
+    Guid? poolId = null,
+    Guid? scalesetId = null,
+    IEnumerable<NodeState>? states = null,
+    PoolName? poolName = null,
+    bool excludeUpdateScheduled = false,
+    int? numResults = null) {
+
+        List<string> queryParts = new();
+
+        if (poolId is not null) {
+            queryParts.Add($"(pool_id eq '{poolId}')");
```

Comment:
should use `TableClient.CreateQueryFilter` so escaping happens properly: see `Queries.cs` for examples

Reply:
will address in follow up PR